src/uwtools/config.py

```diff
@@ -70,6 +73,17 @@ def __init__(self, config_path=None):
 
         self.config_path = config_path
 
+        #initialize logger with parent inheritance
+        if __name__ == '__main__':
+            _name = __name__
+        else:
+            py_caller = getmodule(stack()[1][0])
+            _name = f"{py_caller.__name__}.{__name__}"
+        self.log = Logger(name=_name).get_logger()
```

Comment:
Hmm, so my logic here is to make the logger a child of the parent function's logger if available. It maintains the description of the history in the log by name stamp and inherits the parent's attributes while linking as well, though it does get somewhat convoluted. I'm not sure if getmodule can pull the full tree of modules to join() them?
It would be simpler to just find the parent and reference its logger object if we don't care about tracking the called modules explicitly